File: migration_project/cpp_qt/src/docks/PropertiesDock.cpp
#include "PropertiesDock.h"
#include "../I18n.h"

#include <QFileInfo>
#include <QJsonArray>
#include <QVBoxLayout>
#include <QFormLayout>
#include <QGroupBox>

PropertiesDock::PropertiesDock(QWidget* parent)
    : QWidget(parent)
{
    auto* layout = new QVBoxLayout(this);
    layout->setContentsMargins(8, 8, 8, 8);

    m_label = new QLabel(I18n::instance().tr("properties.none"), this);
    m_label->setAlignment(Qt::AlignCenter);
    m_label->setWordWrap(true);
    layout->addWidget(m_label);
    layout->addStretch();
}

void PropertiesDock::showProject(const QJsonObject& project, const QString& projectPath) {
    auto* lay = qobject_cast<QVBoxLayout*>(layout());
    if (!lay) return;

    QLayoutItem* item;
    while ((item = lay->takeAt(0)) != nullptr) {
        if (item->widget() && item->widget() != m_label)
            item->widget()->deleteLater();
        delete item;
    }

    if (project.isEmpty()) {
        m_label->setText(I18n::instance().tr("properties.none"));
        lay->addWidget(m_label);
        lay->addStretch();
        return;
    }

    m_label->setText(I18n::instance().tr("properties.selection"));
    lay->addWidget(m_label);

    auto* form = new QFormLayout();
    form->addRow(I18n::instance().tr("properties.name"), new QLabel(project.value("project_name").toString()));
    form->addRow(I18n::instance().tr("properties.path"), new QLabel(projectPath));
    form->addRow(I18n::instance().tr("properties.schema"),
                 new QLabel(QString::number(project.value("schema_version").toInt())));
    form->addRow(I18n::instance().tr("properties.modified"),
                 new QLabel(project.value("modified_time").toString()));

    auto* counts = new QFormLayout();
    counts->addRow(I18n::instance().tr("properties.resources"),
                   new QLabel(QString::number(project.value("resources").toArray().size())));
    counts->addRow(I18n::instance().tr("properties.data_sources"),
                   new QLabel(QString::number(project.value("data_sources").toArray().size())));
    counts->addRow(I18n::instance().tr("properties.results"),
                   new QLabel(QString::number(project.value("result_history").toArray().size())));

    lay->addLayout(form);
    lay->addSpacing(12);
    lay->addLayout(counts);
    lay->addStretch();
}

void PropertiesDock::retranslateUi() {
    m_label->setText(I18n::instance().tr("properties.none"));
}

File: migration_project/ui_qt/docks/properties_dock.py
"""Properties panel — shows project and per-resource details."""

from __future__ import annotations

from PySide6.QtWidgets import QLabel, QPlainTextEdit, QVBoxLayout, QWidget

from ..i18n import tr


class PropertiesDock(QWidget):
    def __init__(self, parent=None):
        super().__init__(parent)
        self._project: dict | None = None
        self._project_path: str | None = None
        self._resource: dict | None = None

        self.title = QLabel()
        self.title.setObjectName("AccentText")
        self.details = QPlainTextEdit()
        self.details.setReadOnly(True)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(8, 8, 8, 8)
        layout.setSpacing(8)
        layout.addWidget(self.title)
        layout.addWidget(self.details)
        self.retranslate_ui()

    def show_project(self, project: dict | None, project_path: str | None = None) -> None:
        self._project = project
        self._project_path = project_path
        self._resource = None
        if not project:
            self.details.setPlainText(tr("properties.none"))
            return
        lines = [
            f"{tr('properties.name')}: {project.get('project_name', tr('project.untitled'))}",
            f"{tr('properties.path')}: {project_path or tr('properties.unsaved')}",
            f"{tr('properties.schema')}: {project.get('schema_version', tr('properties.unknown'))}",
            f"{tr('properties.modified')}: {project.get('modified_time', '-')}",
            f"{tr('properties.resources')}: {len(project.get('resources', []))}",
            f"{tr('properties.data_sources')}: {len(project.get('data_sources', []))}",
            f"{tr('properties.results')}: {len(project.get('result_history', []))}",
        ]
        self.details.setPlainText("\n".join(lines))

    def show_resource(self, resource: dict | None) -> None:
        self._resource = resource
        if not resource:
            self.show_project(self._project, self._project_path)
            return
        lines = []
        for key in ["name", "source_path", "source_type", "extension",
                     "size_bytes", "crs", "epsg", "width", "height",
                     "bands", "dtype", "point_count", "face_count", "dimensions",
                     "format_detail", "warning"]:
            val = resource.get(key)
            if val is not None and val != "":
                lines.append(f"{tr(f'properties.{key}')}: {val}")
        if not lines:
            lines.append(tr("properties.none"))
        self.details.setPlainText("\n".join(lines))

    def retranslate_ui(self) -> None:
        self.title.setText(tr("properties.selection"))
        if self._resource:
            self.show_resource(self._resource)
        else:
            self.show_project(self._project, self._project_path)
